skimage/data/__init__.py

```diff
@@ -231,11 +231,15 @@ def _fetch(data_filename):
 
 
 def _init_pooch():
-    os.makedirs(data_dir, exist_ok=True)
+    global data_dir
+    try:
+        os.makedirs(data_dir, exist_ok=True)
+    except PermissionError:
+        data_dir = osp.join(tempfile.gettempdir(), 'skimage_data')
```

Comment:
If I am not mistaken, this is going to download all the data again and again on system reboot. I guess the question is: do we want this or do we rather raise a PermissionError. I am in favour of the latter option.

Reply:
I think that the data are lazy loaded so it is not supposed to be downloaded at system reboot. But if the user uses any image from the `data` module in his code, it will be in fact downloaded at each system reboot.
But I agree with you, this operation must not be pass silently and at least a warning must be raised.
I am not in favor of error raising since this bug prevents the use of `skimage` and `pooch` conjointly even if the `data` module is not used by the user.

Reply:
I'm leaning toward the following proposal:
* `data_dir` always points to `legacy_data_dir`.
    * This retains old behavior as expected
    * You won't be able to use `os.listdir` to find the **new** donwloaded datasets. This was always confusing anyway.
    * We should be able to remove the readme at this stage.
* New datasets, are simply unavailable without pooch.
* Large datasets are simply unavailble if we don't have a place to put them.

Reply:
In this way, directories do not have to be created until we **need** them, and we only **need** it for new, functionality, that doesn't occur at import time.

The old strategy had the advantage of:
* New and old datasets are in the same location.

I don't see a way to have both advantages at once (without seriously hacking and changing the value of `data_dir` from underneath somebody)